HW06.Rmd

```diff
@@ -1,92 +1,174 @@
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+return(ens.mapped_GBM)
+}
+
+#Now, use the function created, for WT and H3K27M mapping for gene names
+WT <- mapping(WT)
+H3K27M <- mapping(H3K27M)
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! I added gene PDK1
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2","PDK1")
+
+
+WT_gene_of_interest <- filter(WT, gene_symbol %in% genes_of_interest)
+WT_gene_of_interest$TPM <- as.numeric(WT_gene_of_interest$TPM)
+
+H3K27M_gene_of_interest <- filter(H3K27M, gene_symbol %in% genes_of_interest)
+H3K27M_gene_of_interest$TPM <- as.numeric(H3K27M_gene_of_interest$TPM)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+
+for(i in 1:27){
+  genes_of_interest[i]
+  WT_loop <- subset(WT_gene_of_interest, gene_symbol == genes_of_interest[i])
+  H3K27M_loop <- subset(H3K27M_gene_of_interest, gene_symbol == genes_of_interest[i])
+  #make sure the t test are in numeric
+  t <- t.test(WT_loop$TPM, H3K27M_loop$TPM)
+  print(t)
+}
+
+#create a readaable summary result for t test
+t <- as.data.frame(t)
+t <- cbind(as.data.frame(WT_gene_of_interest$gene_symbol), t)
+colnames(t) <- c("gene_symbol" , "t_test_for_WT_H3K27M")
+
+#print out the t-test results
+```
+
+
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+
+WT_gene_of_interest[,3] <- c("WT")
+colnames(WT_gene_of_interest) <- c("gene_symbol","TPM","category")
+H3K27M_gene_of_interest[,3] <- c("H3K27M")
+colnames(H3K27M_gene_of_interest) <- c("gene_symbol","TPM","category")
+
+WT_H3K27M <- rbind(WT_gene_of_interest, H3K27M_gene_of_interest)
+
+for(i in 1:27){
+  
+  WT_H3K27M_loop <- subset(WT_H3K27M, gene_symbol == genes_of_interest[i])
+  #make sure the t test are in numeric
+  ggplot(WT_H3K27M_loop, aes(x=category, y=TPM))+
+    geom_boxplot(aes(color = category))+
+    labs(x = "Category", y = "TPM", title = WT_gene_of_interest[i])
+  
+}
+
```

Comment:
Great job! This seemed very challenging. I did not do this assignment (did something with my own data), so it was very helpful to see these solutions! Sorry if my comments are not the most helpful, but I always learn something new from reading my peers code and find things that would be very handy in my own applications! Thanks!